### DISCUSSION TITLE
Странное поведение матрицы с более, чем 100 диодами

### DISCUSSION THREAD

**Discussion by mark-tt** [vvip-68/GyverPanelWiFi#288] [Задавайте вопросы] [answered]
Друзья, доброго врмени суток!

Обыскал все, что только можно, но к сожалению ничего не нашел, посему взываю к помощи умных людей.

Собрал матрицу на окно на адресных светодиодах WS2812 по гайду vvip-68/GyverPanelWIFI на Wemos D1 Mini с реле и SD-ридером (хотя он в коде отключен).

Купил вроде как неплохой блок питания на 10А (ноунейм, но выглядит достаточно легитимно как по мне, фото прикрепляю)

![Screenshot_20231206_234016](https://github.com/vvip-68/GyverPanelWiFi/assets/4976790/e5175b62-95b9-4a4b-8cb3-03fbc2236e71)

Напряжение на блоке выставлено 4.8В.

Спаяно все по соответствующей схеме, единственное чего нет это опциональных конденсаторов, (обязательного?) 200-омного резистора и диода (еще не доехали)

Напряжение на входе гирлянды стабильно 4.7, на выходе в зависимости от режимов от 3.3 до 4.5 (просадка, очевидно).

Собственно в чем проблема - при установке числа диодов в матрице больше 100 (примерно) начинают появляться странные артефакты, обозначенные на видео. До 100 диодов этих артефактов не наболюдается вообще. (Часть диодов горит постоянно одним цветом, при смене цветов бывает меняются на противоположный относительно остальных, 1 или 2 мигают). Причем появляются они не во всех режимах. В режимах, где одновременно активно мало диодов (типа змейка) их нет и все работает отлично.

Видео:

https://youtube.com/shorts/-LVX0Y1sDpk?si=vz9s82bYJsoH5Ti-
https://youtube.com/shorts/qDVibm7E9F8?si=VvmjSaWBUeW1N7D3

Я думал, что проблемы с питанием - не вопрос, впаялся посередине и в конце от того-же блока. Вообще все стало очень странно - все работает также с артефактами, а потом через 10 секунд после старта постепенно угасает до нуля (контроллер думает, что слишком большой ток?). Не помогло.

Также попытался посреди работы режима отключать сигнальный провод - удивительно, но эти ошибочно мигающие диоды переставали мигать (то есть это контроллер подает какой-то ложный сигнал?).

Пытался и так и сяк но ничего не получается. Вдруг есть кто-то с похожими проблемами, был бы крайне признателен за помощь и советы!

**Comment by vvip-68** [comment, marked as answer]
Попробуйте повысить напряжение до 5 вольт, но не более 5.2

Резистор 200 ом - если ничего не сгорело - то можно без него. У меня работают гирлянды без него.
По идее он призван ограничивать максимальный ток через пин контроллера, который управляет лентой, если случайно что-то замкнули или первый диод попался испорченный, чтобы пин не выгорел.

На приложенных видео не вижу никаких проблем. Вроде все работает стабильно. Мигающий диод - скорее всего разделительная точка в часах "отбивает" полусекунды.

До 100 диодов не наблюдается вообще - потому что на этом размере часы просто не показываются.
Или выключите показ часов в приложении.

Разложите ленту в правильную матрицу, установите в приложении правильные соответствующие разложенной матрице размеры, покажите что идет не так. Пока что по видео - у вас все корректно работает.

И видео как "угасает до нуля" тоже приложите.

**Reply by mark-tt** [reply]
Добрый вечер! 

Большое спасибо за детальный ответ!
Действительно, это часы, какой стыд)
Было неочевидно, что они включаются по-умолчанию в каждом режиме... Спасибо!

В схеме ничего не трогал.

Все действительно работает корректно, также оказалось, что эта проблема с угасанием никак не связана с доп питанием - она наболюдается и сейчас, но не всегда. Наблюдение - если перезагрузить питание системы (из розетки) когда реле включено - все окей. А если же выключить реле, перезагрузить питание и потом его включить - будет вот так как на видео.
(Видео угасания вот: https://youtu.be/fgx0RWjJ_mU?si=GgIzKrUJtWlPpcSA
Видно, что остаются в конце очень блеклые часы)

Спасибо вам большое за поддержку и развитие этого замечательного проекта, и еще раз благодарю за умные ответы на тупые вопросы)

**Reply by vvip-68** [reply]
Из описания непонятно - есть ли у вас в cхеме конпка?  И какая точно версия гирлянды?
Из приведенных симптомов можно сделать следующие догадки:
1. Если кнопки нет - ее пин (назначенный в настройках скетча) не притянут к питанию, скетч на нем ловит какие-то наводки, которые воспринимает как сигнал на изменение яркости - в данном случае на уменьшение
2. Если кнопка есть - соответствует ли она типу. Например, в настройках указано, что кнопка тактовая, а фактически стоит сенсорная
3. Соответствует ли версия ядра (рекомендованная 2.7.4) той, на которой собрана прошивка. Версии 3.x.x в отличие от 2.7.4 при неизменных настройках пинов по умолчанию выводят сигнал на ленту на D4 вместо D2 - кнопка назначена на тот же пин и сигнал на ленту воспринимается как сигнал на уменьшение яркости
4. Судя по описание поведения с реле - возможно что в настройках назначение пина управление реле на тот же пин, что и на кнопку. Тогда реле включаясь, устанавливает на пине уровень, который скетчем воспринимается как сигнал кнопки на изменение яркости.
5. Возможно у вас старая версия скетча. Примерно в конце ноября этого года было исправление ошибки - скетч сразу после запуска воспринимал кнопку как нажатую, хотя реально она в обычном состоянии. Из за того, что он считал кнопку нажатой - яркость гасла до нуля вот как у вас. Обновите скетч до последней версии.

Если проблема с пересечением пинов назначения - переназначьте кнопку на пин D6 к примеру.

**Reply by mark-tt** [reply]
Принято, спасибо за ответ, буду разбираться. Кнопки в схеме нет.
Да, у меня действительно гирлянда работает от четвертого пина вместо второго, вероятно и правда он думает что там подключена кнопка.

**Reply by vvip-68** [reply]
Это скорее всего означает, что вы не выполнили условие по версии ядра - 2.7.4 и поставили какое попало. Обычно это из серии 3.x.x. Тогда кроме такого переназначения пина вывода на ленту, падения яркости вы получите еще кучу нежелательных последствий, в виде несистемно горящего первого светодиода, торможения в эффектах, непонятных случайных вспышек и цифровых артефактов. Но это, конечно, ваш выбор...

**Reply by mark-tt** [reply]
Понял, буду исправлять. Инструкцию вроде читал достаточно внимательно, но что-то где-то упустил. Перечитаю.
